tests/Doctrine/Tests/ORM/Functional/QueryTest.php

```diff
@@ -867,4 +867,40 @@ public function testMultipleJoinComponentsUsingLeftJoin()
         $this->assertInstanceOf(CmsUser::class, $users[2]);
         $this->assertNull($users[3]);
     }
+
+    public function testProcessParameterValueWithObjects()
+    {
+        $query = $this->_em->createQuery();
+
+        // non-mapped class
```

Comment:
Split the test in two methods, instead of testing first against a value type, then against an entity type.